### ISSUE TITLE
Drawer without overlay still prevents interaction with the DOM outside the drawer

### ISSUE THREAD

**Issue by fvaldes33** [mantinedev/mantine#2289]
### What package has an issue

@mantine/core

### Describe the bug

Just upgraded from 4.x to the latest 5.x version.

Drawers with `withOverlay: false` before allowed you to interact with the DOM behind the drawer. The drawer root had a height and width of 0px.

Now, even with `withOverlay: false` the drawer root is fixed and inset 0 which prevents clicks from making it to the dom behind.

### What version of @mantine/hooks page do you have in package.json?

5.2

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/embed/misty-haze-8jx2p5

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

The withOverlay prop should be passed into the createStyles method and alter the output of the root styles.

**Comment by nmay231** [comment]
Probably need `withinPortal`, but if that doesn't work, please provide a minimal code example in a codesandbox.

**Comment by fvaldes33** [comment]
@nmay231 - Even with `withinPortal` I still see the same behavior. codesandbox below

https://codesandbox.io/embed/misty-haze-8jx2p5

**Comment by nmay231** [comment]
@fvaldes33 Now I understand. You can use the styles API to get the previous behaviour back: `styles={{ root: { width: 0, height: 0 } }}`.

https://codesandbox.io/s/cocky-torvalds-vbffpz?file=/src/App.tsx:710-752

**Comment by fvaldes33** [comment]
@nmay231 - Totally, that is what I ended up doing but, should this be fixed in the library?

**Comment by nmay231** [comment]
It probably should. It doesn't make sense for that behavior when `withOverlay={false}`. @rtivital ?

**Comment by rtivital** [comment]
Yeah, I think it is a bug, will be fixed in one of next patches, PR is welcome.

**Comment by italodeandra** [comment]
Fixed on https://github.com/mantinedev/mantine/pull/2351
